### PR TITLE
Bigtable:Adding __eq__ and __ne__ methods for row set

### PR DESCRIPTION
**Pull request by sumit-ql**
1. __eq__ and __ne__ methods for RowSet and RowRange.
2. Unit tests.

### CODE REVIEW

File: bigtable/google/cloud/bigtable/row_set.py [hunk 1]
```diff
@@ -29,6 +29,32 @@ def __init__(self):
...
+            return NotImplemented
+
+        if len(other.row_keys) != len(self.row_keys):
+            return False
+
+        if len(other.row_ranges) != len(self.row_ranges):
+            return False
+
+        if not sorted(other.row_keys) == sorted(self.row_keys):
+            return False
```
Comment (tseaver) [inline bigtable/google/cloud/bigtable/row_set.py]: Sorting might be a little expensive.  Can we achieve the same goal by creating sets?  E.g.:

```python
        if not set(other.row_keys) == set(self.row_keys):
            return False
```

File: bigtable/google/cloud/bigtable/row_set.py [hunk 2]
```diff
@@ -29,6 +29,32 @@ def __init__(self):
...
+
+        row_ranges = list(self.row_ranges)
+
+        for each in other.row_ranges:
+            try:
+                row_ranges.remove(each)
+            except ValueError:
+                return False
```
Comment (tseaver) [inline bigtable/google/cloud/bigtable/row_set.py]: This seems more complicated than necessary, and is likely slow (raising / catching exceptions is lots slower than normal code).  I think we can use sets here as well:

```python
        if not set(other.row_ranges) == set(self.row_ranges):
            return False
```

File: bigtable/google/cloud/bigtable/row_set.py (hunk 1 above)
Comment (sumit-ql) [reply bigtable/google/cloud/bigtable/row_set.py]: Yes, we can use the set here. Updated the code.

File: bigtable/google/cloud/bigtable/row_set.py (hunk 2 above)
Comment (sumit-ql) [reply bigtable/google/cloud/bigtable/row_set.py]: Added __hash__() to make RowRange hashable. Now, we can use set while comparing row_ranges.

File: bigtable/google/cloud/bigtable/row_set.py [hunk 3]
```diff
@@ -138,6 +133,24 @@ def __init__(self, start_key=None, end_key=None,
...
+
+        Returns:
+            Tuple[str]: The contents of this :class:`.RowRange`.
+        """
+        return (
+            self.start_key,
+            self.start_inclusive,
+            self.end_key,
+            self.end_inclusive,
+        )
```
Comment (tseaver) [inline bigtable/google/cloud/bigtable/row_set.py]: Would it make sense to use `_key` below in `_eq`, as well?

Comment (sumit-ql) [reply bigtable/google/cloud/bigtable/row_set.py]: Yes. Updated the code, thanks.

**Pull request merged**